apps/order/serializers.py

```diff
@@ -0,0 +1,24 @@
+from rest_framework import serializers
+from django.db.models import F
+
+from apps.order.models import CartItem
+from config.serializers import CreatedUpdatedBaseSerializer
+
+
+class CartItemSerializer(CreatedUpdatedBaseSerializer, serializers.ModelSerializer):
+
+    class Meta:
+        model = CartItem
+        fields = ('book', 'quantity',)
+
+    def create(self, validated_data):
+        created_by = self.context['request'].user
+        validated_data['created_by'] = created_by
+        book = validated_data.get('book', '')
+        quantity = validated_data.get('quantity', '')
+        # Update quantity in cart if book is already in cart
+        if CartItem.objects.filter(created_by=created_by, book=book).exists():
+            return CartItem.objects.filter(created_by=created_by, book=book).update(quantity=F('quantity') + quantity)
+        else:
+            # Otherwise create new cart item
+            return CartItem.objects.create(**validated_data)
```

Comment:
We might not need this if quantity is not required. But for a single instance, let's fetch the object and modify it instead of using `.update(` as it will not make that much of a difference.
```suggestion
       existing_cart_item = CartItem.objects.filter(created_by=created_by, book=book).first()
        if existing_cart_item:
            existing_cart_item.quantity =+ quantity
        else:
            # Otherwise create new cart item
            return CartItem(**validated_data)
        existing_cart_item.save()
        return existing_cart_item
```